cmd/cat/main.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+)
+
+func main() {
+	var nFlag bool
+	flag.BoolVar(&nFlag, "n", false, "to number output lines")
+
+	flag.Usage = func() {
+		fmt.Println("Usage: cat [options] [file]")
+		fmt.Println("Options:")
+		flag.PrintDefaults()
+	}
+
+	flag.Parse()
+
+	filePath := flag.Arg(0)
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		log.Fatalf("Error happened: %v", err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	lineCnt := 1
+	for scanner.Scan() {
+		if nFlag {
+			fmt.Printf("%d ", lineCnt)
+		}
+		fmt.Println(scanner.Text())
+		lineCnt++
+	}
```

Comment:
for consistency with other projects, move any functionality to `internal/` directory and leave main() for only parsing flags and args